### PR TITLE
RS School: AWS Task 6

### PR DESCRIPTION
**Pull request by kagafon**
**Implemented in product-service:**
 - File serverless.ts contains configuration for catalogBatchProcess function
 - File serverless.ts contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS
 - File serverless.ts contains configuration for SQS catalogItemsQueue
 - File serverless.ts contains configuration for SNS Topic createProductTopic and email subscription
 - catalogBatchProcess lambda is covered by unit test
 - Set a Filter Policy for SNS createProductTopic in serverless.ts

Total: **6**

FE link:
https://d1mi29n8b3y4rd.cloudfront.net/

CSV example:
https://github.com/kagafon/nodejs-aws-be/blob/task-6/import-service/data/example.csv

### CODE REVIEW

File: product-service/__tests__/handlers/getProductById.test.ts [hunk 1]
```diff
@@ -1,46 +0,0 @@
-import axios from 'axios';
```
Comment (kirill-ianchuk) [inline product-service/__tests__/handlers/getProductById.test.ts:1]: Why did you delete this test suite?

File: import-service/handlers/importFileParser.ts [hunk 2]
```diff
@@ -25,6 +28,16 @@ const importFileParser = async (
                   let record = parser.read();
                   while (record) {
                     console.log(record);
+                    console.log(process.env.IMPORT_SQS);
+                    sqs.sendMessage(
```
Comment (kirill-ianchuk) [inline import-service/handlers/importFileParser.ts:32]: How do you control the amount of sendMessage requests being in flight at the same time? (in case csv file is huge and scv parser produces data fast)

File: product-service/handlers/catalogBatchProcess.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  client.connect();
+
+  try {
+    await client.query('BEGIN');
+    let maxProductPrice = -1;
+
+    for (const record of event.Records) {
+      const { title, description, price, count } = JSON.parse(record.body);
+      const priceAsNum = +price;
+      const countAsNum = +count;
```
Comment (kirill-ianchuk) [inline product-service/handlers/catalogBatchProcess.ts:19]: You can convert all numeric strings to numbers in the csv parser: https://github.com/mafintosh/csv-parser/issues/49#issuecomment-414070999

File: product-service/serverless.ts [hunk 4]
```diff
@@ -3,15 +3,15 @@ import type { Serverless } from 'serverless/aws';
...
-    // org: your-org-name,
   },
+
   frameworkVersion: '2',
   custom: {
     webpack: {
       webpackConfig: './webpack.config.js',
-      includeModules: true,
+      includeModules: {
+        forceExclude: 'aws-sdk',
```
Comment (kirill-ianchuk) [inline product-service/serverless.ts:13]: Nice

File: product-service/__tests__/handlers/getProductById.test.ts (hunk 1 above)
Comment (kagafon) [reply product-service/__tests__/handlers/getProductById.test.ts:1]: It was implemented for local data files and is not working with PostgreSQL

File: import-service/handlers/importFileParser.ts (hunk 2 above)
Comment (kagafon) [reply import-service/handlers/importFileParser.ts:32]: Never think about it. Does it increase costs?

Comment (kirill-ianchuk) [reply import-service/handlers/importFileParser.ts:32]: Theoretically might be some issues with the amount of callbacks being kept in the memory
